### PR TITLE
A3IDES-676 move eeprom settings to wui thread

### PR DESCRIPTION
**Pull request by michalrudolf**
EEPROM setting moved to WUI thread. While testing I found a bug, when queue get filled up => fixed.

### CODE REVIEW

File: src/wui/wui.c [hunk 1]
```diff
@@ -119,7 +122,17 @@ static void wui_queue_cycle() {
 }
 static int process_wui_request() {
 
-    //if(wui.request == gcode)
-    marlin_json_gcode(wui.request);
+    if(strncmp(wui.request, "!cip ", 5) == 0){
+        uint32_t ip;
+        if(sscanf(wui.request + 5, "%u", &ip)){
+            eeprom_set_var(EEVAR_CONNECT_IP, variant8_ui32(ip));
```
Comment (joshymjose) [inline src/wui/wui.c:127]: setting to eeprom and checking of the IP and other condition should be done only at one place for this and ini file

File: src/wui/wui.c [hunk 2]
```diff
@@ -12,9 +12,12 @@
 #include "lwip.h"
 #include "ethernetif.h"
 #include "http_client.h"
+#include "eeprom.h"
+#include <string.h>
 
 #define MAX_WUI_REQUEST_LEN       100
 #define MAX_MARLIN_REQUEST_LEN    100
+#define MAX_IP4_ADDR_SIZE         16
```
Comment (joshymjose) [inline src/wui/wui.c]: where do you use this?

File: src/wui/wui.c (hunk 1 above)
Comment (michalrudolf) [reply src/wui/wui.c:127]: Agreed. This will be implemented when ini file handling will be seperated from GUI.

File: src/wui/wui.c (hunk 2 above)
Comment (michalrudolf) [reply src/wui/wui.c]: aaa, thats a leftover..

**Review by joshymjose — DISMISSED** [review]
try to make EEPROM operation and error check on IP etc common for GUI and WUI

**Review by joshymjose — APPROVED** [review]

**Pull request merged**